### PR TITLE
[ci] Reset the cache before each rspec suite

### PR DESCRIPTION
**Pull request by coolo**
With things like database_cleaner you really want to make sure that
tests don't pollute each other with stale cache items

(rails defaults to null_store in test environment, but that's a
harsher measure)

### CODE REVIEW

File: src/api/spec/support/cache.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# reset the cache before each test
+RSpec.configure do |config|
+  config.before(:suite) do
```
Comment (ChrisBr) [inline src/api/spec/support/cache.rb]: Before :suite executes this before running all tests, not before every test. If you want to execute it before every test you need ``config.before(:each) ...``

Comment (coolo) [reply src/api/spec/support/cache.rb]: that makes sense :)

**Comment by hennevogel** [comment]
Why not use :null_store if you're sure that the Relationship stuff works without cache?

**Comment by coolo** [comment]
I'm not sure it works without cache. Actually a cache *within* one scenario sounds like a good thing to have

**Comment by coolo** [comment]
but we can try in a 2nd round

**Comment by coolo** [comment]
@ChrisBr please rereview

**Comment by coolo** [comment]
28 failures with null_store, so better don't: https://travis-ci.org/coolo/open-build-service/jobs/398731908

**Pull request merged**